include/CL/sycl/vec.hpp

```diff
@@ -63,7 +63,7 @@ class vec : public detail::small_array<DataType,
 
 
   // Inherit of all the constructors
-  using typename basic_type::small_array;
+  using basic_type::small_array;
```

Comment:
Thank you for investigating.
I am still b=puzzled by the fact your example includes `<iostream>` but uses `printf()`. Strangely GCC is OK with this, but Clang fails for good reasons. Can you update the example? You can select Clang trunk to use the latest version.

Thanks.